HealthcareSystem/PatientWebApp/wwwroot/js/patient_examinations.js

```diff
@@ -386,4 +328,102 @@ function cancelExamination(id) {
 
 function FillOutTheSurvey(examinationId) {
     window.location.href = '/html/filling_out_the_survey.html?id=' + examinationId;
+};
+
+function step(id) {
+
+    if (id == 1) {
+        document.getElementById("div_date").style.display = "none";
+        document.getElementById("div_specialty").style.display = "initial";
+    }
+    if (id == 2) {
+        document.getElementById("div_specialty").style.display = "none";
+        document.getElementById("div_doctor").style.display = "initial";
+    }
+    if (id == 3) {
+
+        var select = document.getElementById("free_appointments");
+        var length = select.options.length;
+        for (i = length - 1; i >= 0; i--) {
+            select.options[i] = null;
+        }
+
+        let date = $('#dateOfExam').val();
+        let doctorJmbg = $('#doctor_name option:selected').val();
+        let specialtyId = $('#specialty_name option:selected').val();
+
+        if (!date || !doctorJmbg || !specialtyId) {
+            let alert = $('<div class="alert alert-danger alert-dismissible fade show mb-0 mt-2" role="alert">You must select a date in the first step.'
+                + '<button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>' + '</div >');
+            
+            $('#alertsModal').prepend(alert);
+            return;
+        }
+            //return;
+
+        $('#loadingSchedule').show();
+        $('#free_appointments').empty();
```

Comment:
I would not switch the page on the modal before the loading is finished. Also, I'd disable the buttons except for Previous while loading.

Reply:
I have changed most things at your suggestion except the following which are not clear to me. I think alert that scheduling was successful is located on 111-113 lines on file patient_examination.js.  
"I would not switch the page on the modal before the loading is finished." Also this part is not the clearest to me. What should I do?

Reply:
Oh, so right now, once everything is chosen, you switch to that last step right a way, and an empty combo box is shown (along with that loading alert). I would maybe just stay on the previous step, show the loading alert, disable the Next button so the user knows it's already been clicked, and then switch to the step with the combo box with appointments once the AJAX is done and appointments are shown.